src/DataStructures/LeviCivitaIterator.hpp

```diff
@@ -61,21 +62,78 @@ class LeviCivitaIterator {
     return indexes_[permutation_];
   }
 
   // Return a specific index from the current (multi-index) Index
-  const size_t& operator[](size_t i) {
-    return indexes_[permutation_][i];
-  }
+  const size_t& operator[](size_t i) { return indexes_[permutation_][i]; }
 
   // Return the sign of the Levi-Civita symbol with these indices
-  int sign() const noexcept { return signs_[permutation_]; }
+  int sign() const noexcept { return signs_[permutation_]; };
+
+  // Initialize the indexes
+  static constexpr cpp17::array<cpp17::array<size_t, Dim>, factorial(Dim)>
+  indexes() noexcept {
+    cpp17::array<cpp17::array<size_t, Dim>, factorial(Dim)> indexes{};
+    cpp17::array<size_t, Dim> index{};
+
+    // Add the initial index permutation to the list
+    // Implement this myself, since iota is not constexpr
+    for (size_t i = 0; i < Dim; ++i) {
+      index[i] = i;
+    }
+    indexes[0] = index;
+
+    // std::next_permutation generates the different permuations of index
+    // loop over them to fill in the rest of the permutations in indexes
+    size_t permutation = 1;
+    while (std::next_permutation(index.begin(), index.end())) {
+      indexes[permutation] = index;
+      ++permutation;
+    }
+    return indexes;
+  };
+
+  static constexpr cpp17::array<int, factorial(Dim)> signs() noexcept {
+    cpp17::array<int, factorial(Dim)> signs{};
+
+    // By construction, the sign of the first permutation is +1
+    signs[0] = 1;
+
+    // How do you know whether the corresponding Levi Civita symbol is
+    // +1 or -1? To find out, compute the number, in the factoradic number
+    // system, corresponding to each permutation, and sum the digits. If the sum
+    // is even (odd), the corresponding Levi-Civita symbol will be +1 (-1). This
+    // works because there are Dim! unique permutations of the Levi Civita
+    // symbol indices, so each one can be represented by a
+    // (`Dim`)-digit factorial-number-system number. For more on the
+    // factoradic number system, see
+    // https://en.wikipedia.org/wiki/Factorial_number_system
+    auto factoradic_counter = cpp17::array<size_t, Dim>();
+    for (size_t permutation = 1; permutation < factorial(Dim); ++permutation) {
+      for (size_t i = 0; i < Dim; ++i) {
+        factoradic_counter[i]++;
+        if (factoradic_counter[i] < i + 1) {
+          break;
+        } else {
+          factoradic_counter[i] = 0;
+        }
+      }
+      signs[permutation] =
+          (std::accumulate(factoradic_counter.begin(), factoradic_counter.end(),
+                           size_t(0)) %
+           2) == 0
+              ? 1
+              : -1;
+    }
+    return signs;
+  };
 
  private:
-  cpp17::array<Index<Dim>, factorial(Dim)> indexes_;
-  cpp17::array<int, factorial(Dim)> signs_;
+  cpp17::array<cpp17::array<size_t, Dim>, factorial(Dim)>
+      indexes_ = indexes();
+  cpp17::array<int, factorial(Dim)> signs_ = signs();
```

Comment:
static constexpr for both of these.

Reply:
Working on this, but currently doesn't compile if I static constexpr both of these, because of a bug in my constexpr reimplementation of std::next_iteration that I'm trying to track down.